lib/route.js

```diff
@@ -428,7 +433,8 @@ function buildRouting (options) {
       loggerOpts.serializers = context.logSerializers
     }
     const childLogger = createChildLogger(context, logger, req, id, loggerOpts)
-    childLogger[kDisableRequestLogging] = disableRequestLogging
+    const resolvedDisableRequestLogging = disableRequestLoggingFn ? disableRequestLoggingFn(req) : disableRequestLogging
+    childLogger[kDisableRequestLogging] = resolvedDisableRequestLogging
```

Comment:
Should we better evaluate this once after the `FastifyRequest` has been constructed?

If done in this way, the caller has all the decorations and potential customization available to implement a robust logic. Otherwise it relies on the `IncomingMessage` object.

It should be safe to instantiate before evaluating, as we just depend on the child logger but not on the `kDisableRequestLogging` value at that specific moment.

Reply:
thank you for the idea, I'll try it out!